packages/client/hmi-client/src/components/widgets/tera-chart-settings-panel.vue

```diff
@@ -6,26 +6,82 @@
 				<h4>Chart Settings</h4>
 			</header>
 			<div class="content">
-				<div v-if="annotations !== undefined">
+				<div v-if="chartAnnotations !== undefined" class="annotation-items">
 					<h5>Annotations</h5>
-					<p>Not yet implemented</p>
+					<div v-for="annotation in chartAnnotations" :key="annotation.id" class="annotation-item">
+						{{ annotation.description }}
+						<span class="btn-wrapper">
+							<Button icon="pi pi-trash" rounded text @click="$emit('delete-annotation', annotation.id)" />
+						</span>
+					</div>
+					<div>
+						<Button
+							v-if="!showAnnotationInput"
+							class="p-button-sm p-button-text"
+							icon="pi pi-plus"
+							label="Add annotation"
+							@click="showAnnotationInput = true"
+						/>
+						<tera-input-text
+							v-if="showAnnotationInput"
+							v-model="generateAnnotationQuery"
+							:icon="'pi pi-sparkles'"
+							:placeholder="'What do you want to annotate?'"
+							:disabled="!!isGeneratingAnnotation"
+							@keyup.enter="createAnnotationDebounced"
+							@keyup.esc="cancelGenerateAnnotation"
+						/>
+					</div>
 				</div>
 			</div>
 		</div>
 	</transition>
 </template>
 
 <script setup lang="ts">
+import _ from 'lodash';
+import { ref, computed } from 'vue';
 import Button from 'primevue/button';
 import { ChartSetting } from '@/types/common';
 import { ChartAnnotation } from '@/types/Types';
+import TeraInputText from '@/components/widgets/tera-input-text.vue';
 
-defineProps<{
+const props = defineProps<{
 	activeSettings: ChartSetting | null;
 	annotations?: ChartAnnotation[];
+	generateAnnotation?: (setting: ChartSetting, query: string) => Promise<ChartAnnotation>;
```

Comment:
Can you comment on why we are passing in a functor here as a prop? As oppose to emit the query as an event and let the parent-component handle the generation? Is there a specific advantage you see doing things this way?

Reply:
Hmm, maybe we can talk about this some more - but should be okay for now I think.

I am just a bit wary of passing down a function closure that can hold onto other references, may need to keep an eye on upstream changes, memory leaks ... that type of things.